executor/sort.go

```diff
@@ -143,6 +143,8 @@ func (e *SortExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	} else {
 		// Check whether the one partition is spilled.
 		// If the partition is in memory, use List.GetRow() to get better performance.
+		e.partitionList[0].GetMutex().RLock()
+		defer e.partitionList[0].GetMutex().RUnlock()
```

Comment:
We may introduce a method named `rowContainer.GetChunk`, which read a chunk from disk if it is spilled, or directly get a Chunk from its memory. After that, the mutex could not be exposed and the code below can be cleaner.